test/interactions/keyInteractionTests.ts

```diff
@@ -2,337 +2,213 @@
+          callback = makeKeyCallback();
+        });
+
+        afterEach(function() {
+          if (this.currentTest.state === "passed") {
+            svg.remove();
+          }
+        });
+
+        type KeyTestCallback = {
+          called: boolean;
+          keycode: number;
+          reset: () => void;
+          (keycode: number): void;
+        }
+
+        function makeKeyCallback() {
+          const callback = <KeyTestCallback> function(keycode?: number) {
+            callback.called = true;
+            callback.keycode = keycode;
+          };
+          callback.called = false;
+          callback.reset = () => {
+            callback.called = false;
+            callback.keycode = null;
+          };
+          return callback;
+        }
+
+        function triggerKeyEvent(mouseEvent: string, p: Plottable.Point, target: d3.Selection<void>, keycode: number, options = null) {
+          TestMethods.triggerFakeMouseEvent(mouseEvent, target, p.x, p.y);
+          TestMethods.triggerFakeKeyboardEvent("keydown", target, keycode, options);
+          if (event === "KeyRelease") {
+            TestMethods.triggerFakeKeyboardEvent("keyup", target, keycode, options);
+          }
+        }
+
+        function registerEvent(keycode: number, callback: KeyTestCallback) {
+          if (event === "KeyPress") {
+            return keyInteraction.onKeyPress(keycode, callback);
+          } else {
+            return keyInteraction.onKeyRelease(keycode, callback);
+          }
+        }
+
+        function unregisterEvent(keycode: number, callback: KeyTestCallback) {
+          if (event === "KeyPress") {
+            return keyInteraction.offKeyPress(keycode, callback);
+          } else {
+            return keyInteraction.offKeyRelease(keycode, callback);
+          }
+        }
+
+        it(`fires callback for ${event}`, () => {
+          assert.strictEqual(registerEvent(A_KEY_CODE, callback), keyInteraction,
+            `setting the ${event} callback returns the interaction`);
+
+          registerEvent(A_KEY_CODE, callback);
+          triggerKeyEvent("mouseover", INSIDE_POINT, eventTarget, A_KEY_CODE);
+          assert.isTrue(callback.called, `callback for "a" was called when "a" key was ${ACTION}`);
+          assert.strictEqual(callback.keycode, A_KEY_CODE, "callback passed correct keycode");
+        });
+
+        it(`only fires callback for key that has been ${ACTION}`, () => {
+          const bCallback = makeKeyCallback();
+
+          registerEvent(A_KEY_CODE, callback);
+          registerEvent(B_KEY_CODE, bCallback);
+
+          triggerKeyEvent("mouseover", INSIDE_POINT, eventTarget, A_KEY_CODE);
+          assert.isTrue(callback.called, `callback for "a" was called when "a" key was ${ACTION}`);
+          assert.isFalse(bCallback.called, `callback for "b" was not called when "a" key was ${ACTION}`);
+        });
+
+        it("does not fire callback when the key is pressed outside of the component", () => {
```

Comment:
pressed is correct here, because if a key is pressed when the mouse is inside the component and released when the mouse is outside of the component, the keyRelease should still be called, and not the other way around